src/main/java/za/ac/sun/cs/adversarial/domain/Board.java

```diff
@@ -33,19 +35,30 @@ public Board (int m, int n, int k) {
         return moves;
     }
 
-    public void makeMove (int player, Move move) {
+    /**
+     * Apply a move from the perspective of a given player onto the board.
+     */
+    public void makeMove(int player, Move move) {
         this.board[move.getRow()][move.getColumn()] = player;
     }
 
+    /**
+     * @param move The move to undo
+     */
     public void undoMove(Move move) {
         this.board[move.getRow()][move.getColumn()] = 0;
     }
 
+    /**
+     * Check if the current state is terminal. If not, return the winning player.
+     *
+     * @return -1 if non-terminal, 0 if draw, 1 for player 1 win, 2 for player 2 win
+     */
     public int isTerminal() {
```

Comment:
There is a new test—`TestReferee.smallerRandomAgents`—we can use to pinpoint where the issue is inside of `isTerminal`.